test/unit/files/test_http.py

```diff
@@ -94,3 +95,22 @@ def check_generic_headers(request, **kwargs):
             "This is ftp.gnu.org, the FTP server of the the GNU project.",
             user_context=user_context,
         )
+
+def test_file_source_file_url():
+    with tempfile.NamedTemporaryFile(mode="w") as tf:
+        tf.write("File content returned from a file:// location")
+        tf.flush()
+        test_url = f"file://{tf.name}"
+        user_context = user_context_fixture()
+        file_sources = configured_file_sources(FILE_SOURCES_CONF)
```

Comment:
Shouldn't this fail because `file://` isn't listed in http_file_source_conf.yml ? IOW is it possible that `file://` isn't checked for access permissions ?

Reply:
The last entry in http_file_sources_conf.yml does not define a url_regex, and therefore, will default to the regex here: https://github.com/galaxyproject/galaxy/pull/15794/files#diff-2fd124bd18e524c4c9f580f624bccdaa072a8c1c4f89183d44b62ef6c9fb087dR43

This would restore the original behaviour that was present, where `stream_url_to_file` would fallback to using `urllib.requests.urlopen`, which is what was handling the `file://` protocol:
https://github.com/galaxyproject/galaxy/pull/15497/files#diff-a5dd9919ac0be1712c9cc71a6746910cd2ba261bfa1f89e50c62bca12e96eeb0L69

In addition to that, any user/role permission checks defined on filesources will apply.

Reply:
Right, but we have additional checks in place upstream to remove the `file://` protocol, I'm not sure these are always being applied in the file sources. I also don't know that it is sufficiently clear that
```
- type: http
```
would include the file protocol ?

Reply:
Would this work better if implemented in the `posix` file source instead of `http`? (I still don't have the full picture in my mind).

Reply:
But yeah, the posix handler is not a built-in/automatically activated filesource, it must be explicitly activated via file_sources_conf.yml. And that's for security reasons, so we probably will have to move file:// out of filesources.

Reply:
> But yeah, the posix handler is not a built-in/automatically activated filesource, it must be explicitly activated via file_sources_conf.yml. And that's for security reasons, so we probably will have to move file:// out of filesources.

I guess we could automatically activate the posix handler as non-writable and admin only? That would need an addition of a `requires_admin` attribute to `BaseFilesSource` and small corresponding modifications to `_parse_common_config_opts()`, `to_dict()` and `user_context_required()` methods, but seems feasible/secure?

Reply:
Is something like this what you had in mind? If so, I could add a few more tests, including adding posix as an automatically loaded/built-in filesource.

Reply:
Looks great to me, thanks @nuwang! What do you think @mvdbeek ?